test/integration/scheduler_perf/update.go

```diff
@@ -0,0 +1,165 @@
+	Opcode operationCode
+	// Namespace the object should be updated in. Must be empty for cluster-scoped objects.
+	Namespace string
+	// Path to spec file describing the object to create.
+	// This will be processed with text/template.
+	// .Index will be in the range [0, Count-1] when updating
+	// more than one object. .Count is the total number of objects.
+	TemplatePath string
+	// Count determines how many objects get updated. Defaults to 1 if unset.
+	Count *int
+	// Template parameter for Count.
+	CountParam string
+	// Number of objects to be updated per second.
+	// If set to 0, all objects are updated at once.
+	// Optional
+	UpdatePerSecond int
+	// Internal field of the struct used for caching the mapping.
+	cachedMapping *meta.RESTMapping
+}
+
+var _ runnableOp = &updateAny{}
+
+func (c *updateAny) isValid(allowParameterization bool) error {
+	if c.Opcode != updateAnyOpcode {
+		return fmt.Errorf("invalid opcode %q; expected %q", c.Opcode, createAnyOpcode)
+	}
+	if c.TemplatePath == "" {
+		return fmt.Errorf("TemplatePath must be set")
+	}
+	if c.UpdatePerSecond < 0 {
+		return fmt.Errorf("invalid UpdatePerSecond=%d; should be non-negative", c.UpdatePerSecond)
+	}
+	// The namespace can only be checked during later because we don't know yet
+	// whether the object is namespaced or cluster-scoped.
+	return nil
+}
+
+func (c *updateAny) collectsMetrics() bool {
+	return false
+}
+
+func (c updateAny) patchParams(w *workload) (realOp, error) {
+	if c.CountParam != "" {
+		count, err := w.Params.get(c.CountParam[1:])
+		if err != nil {
+			return nil, err
+		}
+		c.Count = ptr.To(count)
+	}
+	c.cachedMapping = nil
+	return &c, c.isValid(false)
+}
+
+func (c *updateAny) requiredNamespaces() []string {
+	if c.Namespace == "" {
+		return nil
+	}
+	return []string{c.Namespace}
+}
+
+func (c *updateAny) run(tCtx ktesting.TContext) {
+	count := 1
+	if c.Count != nil {
+		count = *c.Count
+	}
+
+	if c.UpdatePerSecond == 0 {
+		for index := 0; index < count; index++ {
+			err := c.update(tCtx, map[string]any{"Index": index, "Count": count})
+			if err != nil {
+				tCtx.Fatalf("Failed to update object: %w", err)
+			}
+		}
+		return
+	}
+
+	ticker := time.NewTicker(time.Second / time.Duration(c.UpdatePerSecond))
+	defer ticker.Stop()
+	for index := 0; index < count; index++ {
+		select {
+		case <-ticker.C:
+			err := c.update(tCtx, map[string]any{"Index": index, "Count": count})
+			if err != nil {
+				tCtx.Fatalf("Failed to update object: %w", err)
+			}
+		case <-tCtx.Done():
+			return
+		}
+	}
+}
+
+func (c *updateAny) update(tCtx ktesting.TContext, env map[string]any) error {
+	var obj *unstructured.Unstructured
+	if err := getSpecFromTextTemplateFile(c.TemplatePath, env, &obj); err != nil {
+		return fmt.Errorf("%s: parsing failed: %w", c.TemplatePath, err)
+	}
+
+	if c.cachedMapping == nil {
+		mapping, err := restMappingFromUnstructuredObj(tCtx, obj)
+		if err != nil {
+			return err
+		}
+		c.cachedMapping = mapping
+	}
```

Comment:
nit: Move this initialization of cachedMapping to around L99.

Reply:
I'm not sure if we can do that when we have template parameters to fill (index and count)